sharktank/sharktank/utils/llm_task.py

```diff
@@ -0,0 +1,167 @@
+import dataclasses
+import iree.runtime
+import numpy
+
+from abc import ABC, abstractmethod
+from typing import Callable, List, Optional, Tuple
+
+import torch
+
+
+def fill_page_table(bs: int, count: int, page_ids: list[list[int]]) -> numpy.ndarray:
+    pages = numpy.zeros((bs, count), dtype=numpy.int64)
+
+    for i, ids in enumerate(page_ids):
+        pages[i, : len(ids)] = ids[:count]
+
+    return pages
+
+
+@dataclasses.dataclass
+class LlmTaskInput:
+    task_id: str
+    tokens: List[int]
+    seq_len: int
+    pages: List[int]
+
+    start_position: Optional[int] = None
+
+
+class LlmTask(ABC):
```

Comment:
I would move this to the scheduler and generalize it.  Most of the basics are pretty use case vague and it would be easy to write some tests on the scheduler + task